subsys/net/lib/lwm2m/lwm2m_rw_oma_tlv.c

```diff
@@ -946,7 +946,8 @@ int do_write_op_tlv(struct lwm2m_message *msg)
 	 * message processing - consecutive blocks will not carry the TLV
 	 * header.
 	 */
-	if (msg->path.obj_id == 5 && msg->path.res_id == 0) {
+	if (msg->in.block_ctx != NULL && msg->in.block_ctx->ctx.current > 0 &&
```

Comment:
How does the message get to this point with a NULL block_ctx or a ctx.current == 0?

Reply:
`block_ctx` field is optional - we only provide a valid `block_ctx` if block transfer was used by the server for a particular transaction (in most cases it's not used, only for firmware updates). So if the block transfer is not used, we can safely parse the TLV header.

But, if the block transfer is used, the actual payload (TLV) will be chunked in multiple messages. The point is, that in such a case the TLV header will only be present in the first message (hence another check for `msg->in.block_ctx->ctx.current > 0`) and not in the consecutive ones. So we need to skip the TLV parsing for blocks other than the initial one.